Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.cpp

```diff
@@ -0,0 +1,564 @@
+/*!
+  \file NodeWiseMeshPartitioner.cpp
+  \date   2016.05
+
+  \brief  Define the members of class NodeWiseMeshPartitioner
+
+  \copyright
+  Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+
+*/
+
+#include "NodeWiseMeshPartitioner.h"
+
+#include <limits>
+#include <iomanip>
+#include <cstdio>  // for binary output
+
+#include <logog/include/logog.hpp>
+
+#include "BaseLib/Error.h"
+
+#include "MeshLib/IO/VtkIO/VtuInterface.h"
+
+#include "MeshLib/Node.h"
+#include "MeshLib/Elements/Element.h"
+
+namespace ApplicationUtils
+{
+struct NodeStruct
+{
+    NodeWiseMeshPartitioner::IntegerType id;
+    double x;
+    double y;
+    double z;
+};
+
+void NodeWiseMeshPartitioner::readMetisData(const std::string& file_name_base)
+{
+    const std::string npartitions_str = std::to_string(_npartitions);
+
+    // Read partitioned mesh data from METIS
+    const std::string fname_parts = file_name_base + ".mesh.npart." + npartitions_str;
+
+    std::ifstream npart_in(fname_parts);
+    if (!npart_in.is_open())
+    {
+        OGS_FATAL(
+            "Error: cannot open file %s. It may not exist! \
+                   \n Run mpmetis beforehand or use option -m",
+            fname_parts.data());
+    }
+
+    const std::size_t nnodes = _mesh->getNumberOfNodes();
+
+    std::size_t counter = 0;
+    while (!npart_in.eof())
+    {
+        npart_in >> _nodes_partition_ids[counter++] >> std::ws;
+        if (counter == nnodes)
```

Comment:
Yes, it discards if there is blank lines. The end of file is checked by (while (!npart_in.eof())).  If the the data in file is less than expected, the status is checked after the loop.
Since the file to be opened is an intermediate one generated by metis, I think, the check is Ok.